src/Pulsar.Client/Api/DeadLetters.fs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Pulsar.Client.Api
+
+open System
+open Pulsar.Client.Api
+open Pulsar.Client.Common
+open System.Collections.Generic
+open System.Runtime.InteropServices
+open FSharp.Control.Tasks.V2.ContextInsensitive
+open FSharp.UMX
+
+type DeadLettersPolicy(maxRedeliveryCount : int, [<Optional; DefaultParameterValue(null : string)>] deadLetterTopic : string) =
+    member __.MaxRedeliveryCount = maxRedeliveryCount
+    member __.DeadLetterTopic = deadLetterTopic
+
+type DeadLettersProcessor
+    (policy: DeadLettersPolicy,
+     getTopicName: unit -> string,
+     getSubscriptionNameName: unit -> string,
+     createProducer: string -> IProducer) =
+
+    let store = Dictionary<MessageId, ResizeArray<Message>>()
+
+    let producer = lazy (
+
+        let topicName =
+            if String.IsNullOrEmpty(policy.DeadLetterTopic) |> not then
+                policy.DeadLetterTopic
+            else
+                (sprintf "%s-%s-DLQ" (getTopicName()) (getSubscriptionNameName()))
+
+        createProducer topicName
+    )
+
+    let sendAndForgetAsync (builder : MessageBuilder) =
```

Comment:
This is not sendAndForget, but send and ignore result =) Should we rename function or change called method?

Reply:
This method SendAsync also can throw an exception

Reply:
Please inspect `ProducerImpl.SendAndForgetAsync`. Body of this method contains similar code, but method named  `SendAndForgetAsync`.

```
member this.SendAndForgetAsync (message: byte[]) =
            task {
                connectionHandler.CheckIfActive()
                let! _ = this.SendMessage (MessageBuilder(message))
                return ()
            }
```

Reply:
Yes, the idea here is that SendMessage returns Task, which is not awaited, so any error will be ignored. In your case the result is not ignored but awaited (even though ignored later), so exception could happen there

Reply:
Aaaa... Ok. May you suggest how to rewrite this method for task and possible exceptions ignorance?